vignettes/pk_adnca.Rmd

```diff
@@ -0,0 +1,928 @@
+library(admiral)
+library(admiral.test)
+library(lubridate)
+library(stringr)
+library(tibble)
+
+data("admiral_adsl")
+data("admiral_ex")
+data("admiral_pc")
+data("admiral_vs")
+
+adsl <- admiral_adsl
+ex <- convert_blanks_to_na(admiral_ex)
+
+# Load PC
+
+pc <- convert_blanks_to_na(admiral_pc)
+
+# Load VS for baseline height and weight
+
+vs <- convert_blanks_to_na(admiral_vs)
+
+# ---- Lookup tables ----
+param_lookup <- tibble::tribble(
+  ~PCTESTCD, ~PARAMCD, ~PARAM, ~PARAMN,
+  "XAN", "XAN", "Pharmacokinetic concentration of Xanomeline", 1,
+  "DOSE", "DOSE", "Xanomeline Patch Dose", 2,
+)
+```
+
+```{r echo=FALSE}
+ex <- filter(ex, USUBJID %in% c("01-701-1028", "01-701-1033", "01-701-1442", "01-714-1288", "01-718-1101"))
+pc <- filter(pc, USUBJID %in% c("01-701-1028", "01-701-1033", "01-701-1442", "01-714-1288", "01-718-1101"))
+```
+
+At this step, it may be useful to join `ADSL` to your `PC` and `EX`
+domains as well. Only the `ADSL` variables used for derivations are
+selected at this step. The rest of the relevant `ADSL` variables will be
+added later.
+
+In this case we will keep `TRTSDT`/`TRTSDTM` for day derivation and
+`TRT01P`/`TRT01A` for planned and actual treatments.
+
+In this segment we will use `derive_vars_merged()` to join the `ADSL`
+variables and the following `{admiral}` functions to derive analysis
+dates, times and days: `derive_vars_dtm()`, `derive_vars_dtm_to_dt`,
+`derive_vars_dtm_to_tm`, `derive_vars_dy`. We will also create `NFRLT`
+for `PC` data based on `PCTPTNUM`. We will create an event ID (`EVID`)
+of 0 for concentration records and 1 for dosing records. This is a
+traditional variable that will provide a handy tool to identify records
+but will be dropped from the final dataset in this example.
+
+```{r eval=TRUE}
+
+adsl_vars <- exprs(TRTSDT, TRTSDTM, TRT01P, TRT01A)
+
+adpc <- pc %>%
+  # Join ADSL with PC (need TRTSDT for ADY derivation)
+  derive_vars_merged(
+    dataset_add = adsl,
+    new_vars = adsl_vars,
+    by_vars = exprs(STUDYID, USUBJID)
+  ) %>%
+  # Derive analysis date/time
+  # Impute missing time to 00:00:00
+  derive_vars_dtm(
+    new_vars_prefix = "A",
+    dtc = PCDTC,
+    date_imputation = "FIRST",
+    time_imputation = "00:00:00",
+    ignore_seconds_flag = FALSE,
+    flag_imputation = "auto"
+  ) %>%
+  # Derive dates and times from date/times
+  derive_vars_dtm_to_dt(exprs(ADTM)) %>%
+  derive_vars_dtm_to_tm(exprs(ADTM)) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  # Derive event ID and nominal relative time from first dose (NFRLT)
+  mutate(
+    EVID = 0,
+    DRUG = PCTEST,
+    NFRLT = if_else(PCTPTNUM < 0, 0, PCTPTNUM), .after = USUBJID
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adpc,
+  display_vars = exprs(
+    USUBJID, PCTEST, ADTM, VISIT, PCTPT, NFRLT
+  )
+)
+```
+
+Next we will also join `ADSL` data with `EX` and derive dates/times.
+This section uses the `{admiral}` functions `derive_vars_merged`,
```

Comment:
include `()` after each function name - won't repeat this comment again, trust you to double check and update any cases